### ISSUE TITLE
Production mode locking table

### ISSUE THREAD

**Issue by sdpurohit** [serge-rider/dbeaver#1510]
Connecting to a MySQL 5.5.49 database using DBeaver 4.0.3 on Mac OS 10.12.3.

My table is being locked by running a select statement.

1. Use mysqldump to dump a table
2. Connect to MySQL in Production mode
3. Run simple select statement 
select * from mydb.sdp_test;
4. Attempt to import the same table using command line
`mysql -umy_username -pmy_password mydb < sdp_test.sql`
**The import hangs.**
5. Query active processes:
```
SELECT * FROM INFORMATION_SCHEMA.PROCESSLIST where command != 'Sleep';
```
```
ID	USER		HOST		DB	COMMAND	TIME	STATE					INFO
1,142	my_username	localhost	mydb	Query	2	Waiting for table metadata lock		DROP TABLE IF EXISTS `sdp_test`
```

**Comment by serge-rider** [comment]
You are in transactional mode (Repeatable Read by default).
Your select locked the table. Perform Commit or Rollback or switch to Auto-Commit mode to release it.

**Comment by sdpurohit** [comment]
Yes, I am in repeatable read. And yes, this is expected when the select is in a transaction. I didn't think that through correctly.

However, the commit and rollback options are greyed out. So I can not commit or rollback.  DBeaver doesn't recognize it is in a transaction until a DML statement has been run.

**Comment by serge-rider** [comment]
Yep. You are right.

From one perspective Commit/Rollback aren't enabled because there is nothing to commit/rollback (there were no changes made). From the other hand there should be a way to release locked tables. And this can be done only by Commit/Rollback/Disconnect.

Probably Commit/Rollback should be enabled after SELECT queries. Or this should be configurable.
Any ideas?

**Comment by sdpurohit** [comment]
I'm not sure I have a good suggestion.  

One thought would be to default Production mode to Read committed, however when I changed to that through the Transaction Mode menu item, in my MySQL instance the select still prevented my import from dropping the table.

**Comment by serge-rider** [comment]
SELECT queries will activate Commit/Rollback in the next version.

**Comment by tdgroot** [comment]
So it should be fixed in 4.0.4?

**Comment by serge-rider** [comment]
Yes. Now SELECT queries triggers Commit/Rollback command activation.
